components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/UserSessionManagementService.java

```diff
@@ -66,7 +69,21 @@ default boolean terminateSessionsByUserId(String userId) throws SessionManagemen
     }
 
     /**
-     * Terminate the session of the given id
+     * Get a specific session of the given user id.
+     *
+     * @param userId    unique id of the user
+     * @param sessionId unique id of the session
+     * @return user session object. In the default method, null is returned.
+     * @throws SessionManagementException if the session retrieval fails
+     */
+    default Optional<UserSession> getSessionBySessionId(String userId, String sessionId)
```

Comment:
Shall we start the param descriptions with an upper case letter and end with a `.`? Shall we fix this in all places?

Reply:
Usually we follow this approach for variable descriptions, error descriptions, return type descriptions.

Reply:
Javadoc updated as per suggested.